ros2/content.md

```diff
@@ -0,0 +1,265 @@
+
+Alternatively, more permissive network setting can be use to share all host network interfaces with the container, such as [`host` network driver](https://docs.docker.com/network/host/), simplifying connectivity with external network participants. Be aware however that this removes the networking namespace separation between containers, and can affect the ability of DDS participants communicate between containers, as documented [here](https://community.rti.com/kb/how-use-rti-connext-dds-communicate-across-docker-containers-using-host-driver).
+
+## Deployment example
+
+### Docker Compose
+
+In this example we'll demonstrate using [`docker-compose`](https://docs.docker.com/compose/) to spawn a pair of message publisher and subscriber nodes in seperate containers connected through shared software defined network.
+
+> Create the directory `~/ros2_demos` and add the first `Dockerfile` example from above. In the same directory, also create file `docker-compose.yml` with the following that runs a C++ publisher with a Python subscriber:
+
+```yaml
+version: '3'
+
+services:
+  talker:
+    build: ./Dockerfile
+    command: ros2 run demo_nodes_cpp talker
+
+  listener:
+    build: ./Dockerfile
+    command: ros2 run demo_nodes_py listener
+```
+
+> Use docker-copose inside the same directory to launch our ROS nodes. Given the containers created derive from the same docker compose project, they will coexist on shared project network:
+
+```console
+$ docker-compose up -d
+```
+
+> Notice that a new network named `ros2_demos` has been created, as can be shown further with:
+
+```console
+$ docker network inspect ros2_demos
+```
+
+> We can monitor the logged output of each container, such as the listener node like so:
+
+```console
+$ docker-compose logs listener
+```
+
+> Finally, we can stop and remove all the relevant containers using docker-compose from the same directory:
+
+```console
+$ docker-compose stop
+$ docker-compose rm
+```
+
+> Note: the auto-generated network, `ros2_demos`, will persist until you explicitly remove it using `docker-compose down`.
+
+
+### Securing ROS 2
+
+Lets build upon the example above by adding authenticated encryption to the message transport. This is done by leveraging [Secure DDS](https://www.omg.org/spec/DDS-SECURITY). We'll use the same ROS 2 docker image to bootstrap the PKI, CAs, and Digitally Signed files.
+
+> Create a script at `~/ros2_demos/keystore/bootstrap_keystore.bash` to bootstrap a keystore and add entries for each node:
+
+``` shell
+#!/usr/bin/env bash
+# Bootstrap ROS 2 keystore
+ros2 security create_keystore ./
+ros2 security create_key ./ talker
+ros2 security create_key ./ listener
```

Comment:
created keystore file system may still be owned by root afterward, perhaps it's worth changing the permission recursively to that of the parent directory, and marking all key material read only.

Reply:
Fixed in 347e32d